docs/how_pool_works.md

```diff
@@ -0,0 +1,189 @@
+The past stays untouched because the already sent funds are out of the sender's control.
+
+# The proxy
+
+The proxy is configured only with a list of weighted receivers.
+It's not possible to send funds via a proxy which isn't configured.
+The sum of the receivers' weights must always be a constant value,
+which is defined in the contract and it's the same for all the proxies.
+
+Just like a receiver, the proxy has a mapping between the cycles and the deltas of received amounts.
+
+## Sending via a proxy
+
+When a sender starts sending funds to a proxy, it does so in two steps.
+First it applies changes to the proxy's deltas similarly to how it would do with a receiver.
+Next, it iterates over all the proxy's receivers and applies changes to
+their deltas as if they were directly funded by the sender.
+The funding rate applied to the receivers is split according to their weights in the proxy.
+
+For the example sake let's assume that the delta's proxy weights sum must be equal 5.
+The proxy has 2 receivers: A with weight 3 and B with weight 2.
+The sender wants to start sending via the proxy 2 per block or 10 per cycle.
+It's 2 per cycle per proxy weight.
+
+![](how_pool_works_6.png)
+
+The proxy's deltas store amount per 1 proxy weight, which is 2.
+The receivers get their shares, A with weight 3 gets 6 and B with weight 2 gets 4 per cycle.
+
+When a sender stops sending, the process is reversed like with regular receivers.
+All the deltas are once again applied, but with negative values.
+
+## Updating a proxy
+
+When the list of proxy receivers is updated, all funding must be moved to a new set of receivers.
+That's when the proxy's deltas come useful.
+For each cycle and each receiver, the proxy can tell, what total delta it has applied.
+It can then use this information to erase its influence from its receivers.
+
+![](how_pool_works_7.png)
+
+In this example, the receiver's weight is 3.
+To erase the influence, the proxy's deltas are multiplied by
+the receiver's weights and subtracted from the corresponding receiver's deltas.
+
+After removing its influence from one set of receivers the proxy must reapply itself on a new set.
+This is done in the same way as removal, but this time the deltas are added and not subtracted.
+
+### The current cycle problem
+
+Unlike the senders, the proxies store data with a per-cycle precision.
+When changing the set of receivers, a delta describing the current cycle may need to be applied.
+When it happens, it's unclear, what part of the per-cycle delta should be moved,
+because some funds were sent before the current block and some will be sent after it.
+
+![](how_pool_works_8.png)
+
+The solution is to ignore the problem and move the whole current cycle delta.
+Some funds already sent in the current cycle may disappear from one receiver and appear in another,
+but it's not dangerous, the receivers have no access to funds coming from an unfinished cycle.
```

Comment:
```suggestion
Some funds already sent in the current cycle may disappear from one receiver and appear in another.
Such behaviour, however, is not of significant importance since the receivers have no access to funds 
coming from an unfinished cycle.
```